docs/src/components/landing-page/MainLayoutHeader.vue

```diff
@@ -1,119 +1,249 @@
+            alt="Quasar Logo"
+            height="48"
+            width="48">
+          <img
+            v-else
+            :height="$q.screen.xs? '24':'48'"
+            :src="`https://cdn.quasar.dev/logo-v2/svg/logo-horizontal${!dark? '-dark':''}.svg`"
+            alt="Quasar Logo"
+            width="236">
+          <q-separator
+            v-if="$q.screen.gt.xs"
+            :color="dark? 'black-12':'lp-primary'"
+            class="q-ml-lg"
+            vertical/>
+        </div>
 
-      <div class="row items-center text-size-16">
-        <div v-if="$q.screen.gt.xs && showNavItems" class="toolbar-menu-items">
-          <q-btn
-              v-for="(navItem, navItemIndex) in navItems.mainNavItems" :key="navItemIndex"
-              :label="navItem.label"
-              :to="computeRouteNav(navItem)"
-              :href="computeRouteNav(navItem, 'href')"
-              :target="navItem.href? '_blank':'_self'"
-              :color="setActivePrimaryNavColor($route.path, navItem.path)"
+        <div class="row items-center">
+          <div
+            v-if="$q.screen.gt.xs && showNavItems"
+            class="toolbar-menu-items">
+            <header-nav-link
+              v-for="(navItem, navItemIndex) in navItems.mainNavItems"
+              :key="navItemIndex"
+              :dark="dark"
+              :nav-item="navItem"
               :padding="`xs ${$q.screen.name}`"
-              class="text-weight-bold text-size-16"
-              flat
-          >
-            <q-menu v-if="navItem.subMenu" class="shadow-bottom-small">
-              <nav-dropdown-menu :nav-items="navItem.subMenu"/>
-            </q-menu>
-          </q-btn>
-        </div>
-        <q-form v-if="isSearchFieldActive" autocapitalize="off" autocomplete="off" spellcheck="false" class="search-form position-relative">
-          <q-input ref="searchInputRef" autofocus v-model="searchTerms" :dark="!dark" dense square debounce="300" @keydown="onSearchKeydown" @focus="onSearchFocus" @blur="closeSearchForm" placeholder="Search Quasar v2...">
-            <template #append>
-              <q-icon v-if="!searchTerms" name="search" size="sm" color="lp-primary" />
-              <q-icon v-else name="cancel" @click.stop="resetSearch" class="cursor-pointer"/>
-            </template>
-          </q-input>
-          <div class="search-result-field">
-            <q-scroll-area dark class="bg-dark text-white rounded-borders search-result-container">
-              <template v-if="searchResults !== null">
-                <component v-if="searchResults.masterComponent !== void 0" :is="searchResults.masterComponent"/>
-                <app-search-results v-else :results="searchResults" :search-has-focus="searchHasFocus" :search-active-id="searchActiveId"/>
-              </template>
-            </q-scroll-area>
+              nav-item-class="text-size-16"
+            />
+          </div>
+          <div ref="searchForm">
```

Comment:
The wrapper div is necessary since we use it to know the region which does not cause the form to close. It wraps the search input field and the icon. Since this hiding and display of the search form is not the business of the search component itself, I preferred to move the wrapper to the parent.